fk/secretreceive_test.go

```diff
@@ -194,5 +203,73 @@ func TestDecryptAPISecret(t *testing.T) {
 			assert.ErrorIsNil(t, err)
 			assert.Equal(t, decryptedSecret.UUID, uuid)
 		})
+
+		t.Run("with an empty filename", func(t *testing.T) {
+			assert.Equal(t, decryptedSecret.filename, "")
+		})
+	})
+
+	t.Run("populates decrypted secret from a file", func(t *testing.T) {
+		mockPrivateKey := &mockDecryptor{
+			decryptedArmoredResult: strings.NewReader(
+				`{"secretUuid": "93d5ac5b-74e5-4f87-b117-b8d7576395d8"}`,
+			),
+			decryptedArmoredToStringResult: "decrypted content",
+			decryptedArmoredToStringLiteralData: &packet.LiteralData{
+				FileName: "example.txt",
+			},
+		}
+		decryptedSecret, err := decryptAPISecret(encryptedSecret, mockPrivateKey)
+		assert.ErrorIsNil(t, err)
+
+		t.Run("with decrypted content", func(t *testing.T) {
+			assert.Equal(t, decryptedSecret.decryptedContent, "decrypted content")
+		})
+
+		t.Run("with decrypted uuid", func(t *testing.T) {
+			uuid, err := uuid.FromString("93d5ac5b-74e5-4f87-b117-b8d7576395d8")
+			assert.ErrorIsNil(t, err)
+			assert.Equal(t, decryptedSecret.UUID, uuid)
+		})
+
+		t.Run("with a matching filename", func(t *testing.T) {
+			assert.Equal(t, decryptedSecret.filename, "example.txt")
+		})
+	})
+}
+
+func mockDoesntExist(path string) bool {
+	if path == "/fake/new_filename.txt" {
+		return true
+	}
+
+	if path == "/fake/existing_filename.txt" || path == "/fake/existing_filename(1).txt" {
+		return false
+	}
+
+	if path == "/fake/old_filename.txt.bak" {
+		return false
+	}
+
+	return true
+}
+
+func TestGetNewUniqueFilename(t *testing.T) {
```

Comment:
Recommend adding a test "gives up if it can't find a filename after 10 attempts"